pkg/component/runtime/service.go

```diff
@@ -136,6 +137,11 @@ func (s *serviceRuntime) Run(ctx context.Context, comm Communicator) (err error)
 	defer cisStop()
 
 	onStop := func(am actionMode) {
+		if stopping {
+			s.log.Debugf("service %s is already stopping: skipping...")
+			return
+		}
+		stopping = true
```

Comment:
That this is never set to false doesn't seem right. What would happen if we stopped the service and then tried to start it again? This would be uninstalling the endpoint integration and then reinstalling it quickly, or perhaps reassigning the agent to a different policy.

The `ignoreCheckins` on this path is set to false again in the `case actionStart:` block.

Reply:
When discussing the fix with @blakerouse , I asked about recycling a service runtime object and we came to the conclusion that a stopped component is ultimately discarded (removed from the maps of running components in the runtime manager) after it eventually stops.

If this assumption does not hold, then we need to keep track of the installation status and if the object gets recycled we need to wait for the ongoing uninstall to finish, then reinstall it (it is done as part of the start()) and set up the component back properly.

@blakerouse any thoughts ?

Reply:
It is correct. Once a manager is stopped it is never restarted again. This follows the same pattern -  https://github.com/elastic/elastic-agent/blob/main/pkg/component/runtime/runtime.go#L190; once set it is not unset.

Reply:
If this is the case we should add a comment explaining this.